client/Crouch.lua

```diff
@@ -96,62 +127,346 @@ local function StartCrouch()
+                        TaskPlayAnimAdvanced(playerPed, "move_crawlprone2crawlfront", "left", playerCoords.x, playerCoords.y, playerCoords.z, 0.0, 0.0, GetEntityHeading(playerPed), 2.0, 2.0, -1, 2, 0.1, false, false)
+                        ChangeHeadingSmooth(playerPed, -10.0, 300)
+                        Wait(700)
+                    else
+                        PlayAnimOnce(playerPed, "get_up@directional_sweep@combat@pistol@left", "left_to_prone", -1)
+                        ChangeHeadingSmooth(playerPed, 25.0, 400)
+                        PlayIdleCrawlAnim(playerPed)
+                        Wait(600)
+                    end
+                    inAction = false
+                end
+            elseif IsControlPressed(0, 35) then -- INPUT_MOVE_RIGHT_ONLY
+                if isCrawling then
+                    local headingDiff = backwards and 1.0 or -1.0
+                    SetEntityHeading(playerPed, GetEntityHeading(playerPed) + headingDiff)
+                else
+                    inAction = true
+                    if proneType == "onfront" then
+                        local playerCoords = GetEntityCoords(playerPed)
+                        TaskPlayAnimAdvanced(playerPed, "move_crawlprone2crawlfront", "right", playerCoords.x, playerCoords.y, playerCoords.z, 0.0, 0.0, GetEntityHeading(playerPed), 2.0, 2.0, -1, 2, 0.1, false, false)
+                        ChangeHeadingSmooth(playerPed, 10.0, 300)
+                        Wait(700)
+                    else
+                        PlayAnimOnce(playerPed, "get_up@directional_sweep@combat@pistol@right", "right_to_prone", -1)
+                        ChangeHeadingSmooth(playerPed, -25.0, 400)
+                        PlayIdleCrawlAnim(playerPed)
+                        Wait(600)
+                    end
+                    inAction = false
+                end
+            end
+
+            if not isCrawling then
+                if IsControlPressed(0, 22) then
+                    CrawlFlip(playerPed)
+                end
+            end
+
+            Wait(0)
+        end
+
+        -- If the crawling wasn't forcefully ended, then play the get up animations
+        PlayExitCrawlAnims(forceEnd)
+
+        -- Reset variabels
+        isCrawling = false
+        inAction = false
+        proneType = "onfront"
+        SetPedConfigFlag(PlayerPedId(), 48, false) -- CPED_CONFIG_FLAG_BlockWeaponSwitching
+
+        -- Unload animation dictionaries
+        RemoveAnimDict("move_crawl")
+        RemoveAnimDict("move_crawlprone2crawlfront")
+    end)
+end
+
+local function CrawlKeyPressed()
+    -- If we already are doing something, then don't continue
+    if inAction then
+        return
+    end
+
+    -- If already prone, then stop
+    if IsProne then
+        IsProne = false
+        return
+    end
+
+    -- If we are crouching we should stop that first
+    local wasCrouched = false
+    if isCrouched then
+        isCrouched = false
+        wasCrouched = true
+    end
+
```

Comment:
we probably don't want to crawl with props attached, so this might be a solution
```suggestion
EmoteCancel()
```